ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionLoaderUtils.java

```diff
@@ -177,14 +181,31 @@ static SchemaProvider handleUdfReturnSchema(
       final Map<GenericType, SqlType> genericMapping = new HashMap<>();
       for (int i = 0; i < Math.min(parameters.size(), arguments.size()); i++) {
         final ParamType schema = parameters.get(i);
```

Comment:
I looked into changing this to also accept a `List<SqlArgument>` but it seemed borderline impossible as it involved changing the schema provider which caused somewhat of a chain reaction of other necessary changes. This way of checking inputs seemingly works although isn't very pretty. Open to ideas for how to improve this